queue_processors/queue_processor/handle_message.py

```diff
@@ -0,0 +1,388 @@
+        run_version = self._get_last_run_version(run_no, experiment=experiment)
+        run_version += 1
+        message.run_version = run_version
+
+        # Get the script text for the current instrument. If the script text
+        # is null then send to
+        # error queue
+        script_text = self._utils. \
+            instrument_variable.get_current_script_text(instrument.name)[0]
+        if script_text is None:
+            self.reduction_error(message)
+            raise InvalidStateException(
+                "Script text for current instrument is null")
+
+        # Make the new reduction run with the information collected so far
+        # and add it into the database
+        reduction_run = self._create_reduction_run_record(
+            experiment=experiment, instrument=instrument, message=message,
+            run_version=run_version, script_text=script_text, status=status)
+        db_access.save_record(reduction_run)
+
+        # Create a new data location entry which has a foreign key linking
+        # it to the current
+        # reduction run. The file path itself will point to a datafile
+        # (e.g. "\isis\inst$\NDXWISH\Instrument\data\cycle_17_1\WISH00038774
+        # .nxs")
+        data_location = self._data_model.DataLocation(file_path=message.data,
+                                                      reduction_run_id=reduction_run.id)
+        db_access.save_record(data_location)
+
+        # We now need to create all of the variables for the run such that
+        # the script can run
+        # through in the desired way
+        self._logger.info('Creating variables for run')
+        variables = self._utils.instrument_variable.create_variables_for_run(
+            reduction_run)
+        if not variables:
+            self._logger.warning(
+                "No instrument variables found on %s for run %s",
+                instrument.name, message.run_number)
+
+        self._logger.info('Getting script and arguments')
+        reduction_script, arguments = self._utils.reduction_run. \
+            get_script_and_arguments(reduction_run)
+        message.reduction_script = reduction_script
+        message.reduction_arguments = arguments
+
+        # Make sure the RB number is valid
+        try:
+            validate_rb_num(message.rb_number)
+        except InvalidStateException as ex:
+            self._construct_and_send_skipped(
+                message=message, rb_number=message.rb_number, reason=str(ex))
+            raise ex
+
+        if instrument.is_paused:
+            self._logger.info("Run %s has been skipped",
+                              message.run_number)
+        else:
+            self._client.send_message('/queue/ReductionPending', message)
+            self._logger.info("Run %s ready for reduction",
+                              message.run_number)
+
+    # pylint: disable=too-many-arguments
+    def _create_reduction_run_record(self, experiment, instrument, message,
```

Comment:
Maybe this could be refactored into the `model.orm.access`?

Reply:
docstring

Reply:
Definitely, I moved it into its own method as I think it shouldn't be here, but wanted to avoid making it difficult to track stuff moving around. I've opened #653 for these

I'm surprised Pylint didn't pick these up, I can't see a `disable=docstring` :thinking: